ros2_batch_job/util.py

```diff
@@ -167,12 +167,28 @@ def warn(*args, **kwargs):
 
 class MyProtocol(AsyncSubprocessProtocol):
     def __init__(self, cmd, exit_on_error, *args, **kwargs):
-        self.cmd = cmd
+        self.cmd = cmd  
         self.exit_on_error = exit_on_error
+        self.progress_bar = False
+        self.progress = []
         AsyncSubprocessProtocol.__init__(self, *args, **kwargs)
 
     def on_stdout_received(self, data):
-        sys.stdout.write(data.decode('utf-8', 'replace').replace(os.linesep, '\n'))
+        if b'[?25l' in data:
+            self.progress_bar = True
+        if b'[?25h' in data:
+            if self.progress:
+                for display in self.progress:
+                    sys.stdout.write(display.decode('utf-8', 'replace').replace(os.linesep, '\n'))
+            self.progress_bar = False
+        if self.progress_bar:
+            self.progress.append(data)
+            if len(self.progress) == 10:
```

Comment:
Do you have a screenshot of what this ends up looking like?  Since we are now only printing every 10th piece of output, I feel like this is going to look pretty weird in the end, but I'm not sure.

Reply:
![image](https://user-images.githubusercontent.com/47810786/84667274-b52d9b00-af54-11ea-9a48-c0cbe94d3dfd.png)
This is a part of pip install process. If it is less than 10, the last few lines will be printed.

Reply:
OK, yeah.  It is slightly weird in that the end `|` moves around, but I think that is an acceptable tradeoff here.  Thanks!